readthedocs/api/v3/permissions.py

```diff
@@ -22,7 +22,9 @@ def has_permission(self, request, view):
                 # hitting ``/projects/``, allowing
                 return True
 
-            if view.detail:
+            # detail view is only allowed on list/retrieve actions (not
+            # ``update`` or ``partial_update``)
+            if view.detail and view.action in ('list', 'retrieve', 'superproject'):
```

Comment:
These feel like they should be constants? What is a `superproject`?  It still feels weird that it is a method name to me.

Reply:
`superproject` is an action name, defined by the class method under `ProjectViewSet`. We should apply the same permissions restrictions than for a detail action (since it only returns _one_ superproject if exists).

`list` and `retrieve` are DRF standard action names (same as `update` or `partial_update`).

We already talked about this at https://github.com/readthedocs/readthedocs.org/pull/5857#discussion_r299172350

(I'm adding this comment as a code comment)